packages/php-db-import-export/tests/functional/Bigquery/ToFinal/IncrementalImportTest.php

```diff
@@ -258,4 +261,191 @@ public function testIncrementalImport(
+                    'price'=> 200,
+                    'isDeleted'=> 0,
+                    '_timestamp'=> '2021-01-01 00:00:00',
+                ],
+                [
+                    'id'=> 3,
+                    'name'=> 'test3',
+                    'price'=> 300,
+                    'isDeleted'=> 0,
+                    '_timestamp'=> '2021-01-01 00:00:00', // no change no timestamp update
+                ],
+                [
+                    'id'=> 4,
+                    'name'=> 'test4',
+                    'price'=> 400,
+                    'isDeleted'=> 0,
+                    '_timestamp'=> '2022-02-02 00:00:00',
+                ],
+            ],
+        ];
+
+        yield 'import typed table, timestamp update onchange feature: `native-types_timestamp-bc`' => [
+            'features' => ['native-types_timestamp-bc'],
+            'expectedContent' => [
+                [
+                    'id'=> 1,
+                    'name'=> 'change',
+                    'price'=> 100,
+                    'isDeleted'=> 0,
+                    '_timestamp'=> '2022-02-02 00:00:00',
+                ],
+                [
+                    'id'=> 2,
+                    'name'=> 'test2',
+                    'price'=> 200,
+                    'isDeleted'=> 0,
+                    '_timestamp'=> '2021-01-01 00:00:00',
+                ],
+                [
+                    'id'=> 3,
+                    'name'=> 'test3',
+                    'price'=> 300,
+                    'isDeleted'=> 0,
+                    '_timestamp'=> '2021-01-01 00:00:00', // no change no timestamp update
+                ],
+                [
+                    'id'=> 4,
+                    'name'=> 'test4',
+                    'price'=> 400,
+                    'isDeleted'=> 0,
+                    '_timestamp'=> '2022-02-02 00:00:00',
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider incrementalImportTimestampBehavior
+     * @param string[] $features
+     * @param array<mixed> $expectedContent
+     */
+    public function testImportTimestampBehavior(array $features, array $expectedContent): void
+    {
+        $this->bqClient->runQuery($this->bqClient->query(sprintf(
+            'CREATE TABLE %s.%s
+            (
+              `id` INT64 NOT NULL,
+              `name` STRING(50),
+              `price` DECIMAL,
+              `isDeleted` INT64,
+              `_timestamp` TIMESTAMP
+           )',
+            BigqueryQuote::quoteSingleIdentifier($this->getDestinationDbName()),
+            BigqueryQuote::quoteSingleIdentifier(self::TABLE_TRANSLATIONS),
+        )));
+        $this->bqClient->dataset($this->getDestinationDbName())->table(self::TABLE_TRANSLATIONS)->update(
+            [
+                'tableConstraints' => [
+                    'primaryKey' => [
+                        'columns' => 'id',
+                    ],
+                ],
+            ],
+        );
```

Comment:
docela zajímavé v BQ sa mě nějak nepodařilo ten constraint udělat v SQL, je dost možné teda že sem to dělale blbě, ale je mě to asi fuk teď :)